frontend/src/components/StockData/StockData.jsx

```diff
@@ -16,20 +17,28 @@ export default function StockData(props) {
                 setStockNotFound(false);
 
                 const databaseResponse = await axios.get(
-                    `http://localhost:3000/stocks/${ticker}`
+                    `http://localhost:3000/stocks/${ticker}`,
+                    { withCredentials: true }
                 );
-                const stockExistsInDatabase = databaseResponse.data
+
+                const stockExistsInDatabase = databaseResponse.data.stock
                     ? true
                     : false;
 
                 if (stockExistsInDatabase) {
-                    setStockData(databaseResponse.data);
+                    setStockData(databaseResponse.data.stock);
                     return;
                 }
 
-                const stockOverviewUrl = `https://www.alphavantage.co/query?function=OVERVIEW&symbol=${ticker}&apikey=${import.meta.env.VITE_ALPHA}`;
-                const stockPriceUrl = `https://finnhub.io/api/v1/quote?symbol=${ticker}&token=${import.meta.env.VITE_FINNHUB}`;
-                const stockLogoUrl = `https://finnhub.io/api/v1/stock/profile2?symbol=${ticker}&token=${import.meta.env.VITE_FINNHUB}`;
+                const stockOverviewUrl = `https://www.alphavantage.co/query?function=OVERVIEW&symbol=${ticker}&apikey=${
```

Comment:
using a separate util for constants and `URL` API would help clean up this for readability

Reply:
Can you show me an example of how it would look? I looked into the URL api but it looks kind of complex.

Reply:
Here's a good [reference](https://fburl.com/acfrjpgx) I found that discusses URL building with the `URL` API. You can create with `new URL` and use `append` function to add parameters like the stock ticker